tools/shared.py

```diff
@@ -1887,12 +1887,8 @@ def link_lld(args, target, opts=[], lto_level=0):
     for a in Building.llvm_backend_args():
       cmd += ['-mllvm', a]
 
-    # emscripten-wasm-finalize currently depends on the presence of debug
-    # symbols for renaming of the __invoke symbols
-    # TODO(sbc): Re-enable once emscripten-wasm-finalize is fixed or we
-    # no longer need to rename these symbols.
-    # if Settings.DEBUG_LEVEL < 2 and not Settings.PROFILING_FUNCS:
-    #   cmd.append('--strip-debug')
+    if Settings.DEBUG_LEVEL < 2 and not Settings.PROFILING_FUNCS:
+      cmd.append('--strip-debug')
```

Comment:
How does this relate to the names section? We should remove full debug info in debug level 2 as well (but not the names section, and not i profiling-funcs, which is why i ask).

We already pass `--strip` to binaryen, do we need both? (I don't care which one we pass.)

Reply:
This flag will remove the names section and the dwarf debug info sections.

I saw your binaryen change and realized we should probably fix this, it might speed up linking and finalizing marginally.    Mostly just for completeness.

Reply:
Sounds good. We should remove the binaryen `--strip` arguments then.

I realized now that in both cases we don't support the case of having just the names section but not debug info. That will be a problem for profiling (but not debugging). Perhaps we could have a binaryen pass that only strips debug info but not names?